front-end/src/components/Chart.js

```diff
@@ -249,14 +263,104 @@ const Chart = ({ chartInformation }) => {
                     }
                 })(),
                 boost: {
-                    enabled: chartInformation.type === "colour" ? false : true
+                    enabled: chartInformation.type === "colour" ? false : false
                 }
 
             }
         })
         
     }, [parsedData, chartInformation, fileNames])
 
+    const [videoTimestamp, setVideoTimestamp] = useState(0);
+    const [isLoaded, setIsLoaded] = useState(false);
+
+    useEffect(() => {
+        if (!chartInformation.window) return;
+
+        const handleWindowLoad = () => {
+            setIsLoaded(true)
+            console.log("window loaded")
+        }
+
+        chartInformation.window.addEventListener('load', handleWindowLoad)
+
+        return () => {
+            chartInformation.window.removeEventListener('load', handleWindowLoad)
+            setIsLoaded(false)
+            console.log("window unloaded")
+        }
+    }, [chartInformation])
+
+    useEffect(() => {
+        if (!chartInformation.window || !isLoaded) return;
+    
+        const videoElement = chartInformation.window.document.getElementById('video');
+        console.log("videoElement:", videoElement)
+        if (!videoElement) return;
+
+        const handleTimeUpdate = () => {
+            setVideoTimestamp(videoElement.currentTime * 1000);
+        };
+        
+        videoElement.addEventListener('timeupdate', handleTimeUpdate);
+        
+        return () => {
+            videoElement.removeEventListener('timeupdate', handleTimeUpdate);
+        };
+    }, [isLoaded]);
+
+    const [offsets, setOffsets] = useState([])
+
+    useEffect(() => {
+        if (chartInformation.video.fileHeaders.length == 0) return
+        const videoStart = new Date(chartInformation.video.fileHeaders[0]).getTime()
+        
+        const tempOffsets = []
+        chartInformation.files.forEach(file => {
+            const fileStart = new Date(file.columns[0].timespan[0]).getTime() // Unix date of first timestamp in file
+            tempOffsets.push(videoStart - fileStart)
+        })
+
+        setOffsets(tempOffsets)
+    }, [chartInformation])
```

Comment:
Duplicate useEffect on chartInformaton as the one you made above, can we split the logic out into their own functions and call them from the same useEffect?

Reply:
good idea

Reply:
duplicate useEffect has been removed